compiler/testData/diagnostics/tests/enum/typeCompatibility_withExtendedTypeCompatibilityChecker.fir.kt

```diff
@@ -0,0 +1,99 @@
+// !LANGUAGE: +ExtendedTypeCompatibilityChecker
+
+class A<T>
+class AIn<in T>
+class AOut<out T>
+class A2
+
+open class B<T>
+open class BIn<in T>
+open class BOut<out T>
+
+open class C
+open class D
+
+enum class E
+interface I
+
+interface T<T>
+
+open class TSub1 : T<String>
+open class TSub2 : T<Int>
+
+fun foo(
+    string: String, int: Int,
+    strings: List<String>, ints: List<Int>,
+
+    aString: A<String>, aInt: A<Int>,
+    aOutString: A<out String>, aOutInt: A<out Int>,
+    aOutString2: AOut<String>, aOutInt2: AOut<Int>,
+    aInString: A<in String>, aInInt: A<in Int>,
+    aInString2: AIn<String>, aInInt2: AIn<Int>,
+
+    bString: B<String>, bInt: B<Int>,
+    bOutString: B<out String>, bOutInt: B<out Int>,
+    bOutString2: BOut<String>, bOutInt2: BOut<Int>,
+    bInString: B<in String>, bInInt: B<in Int>,
+    bInString2: BIn<String>, bInInt2: BIn<Int>,
+
+    a2: A2,
+
+    e: E,
+    i: I,
+
+    ac: A<C>, ad: A<D>,
+
+    tSub1: TSub1,
+    tSub2: TSub2,
+
+    aListInt: A<List<Int>>,
+    aSetInt: A<Set<Int>>,
+    aListString: A<List<String>>,
+) {
+    "a" == "b"
+    1 == 2
+    <!EQUALITY_NOT_APPLICABLE!>"" == 2<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>string == int<!>
+    <!EQUALITY_NOT_APPLICABLE!>strings == ints<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>aString == aInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>aOutString == aOutInt<!>
+    aInString == aInInt
+    <!EQUALITY_NOT_APPLICABLE!>aOutString == aInInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>aInString == aOutInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>aOutString == aInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>aInString == aInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>aOutString2 == aOutInt2<!>
+    aInString2 == aInInt2
+    <!EQUALITY_NOT_APPLICABLE!>aOutString2 == aInInt2<!>
+    <!EQUALITY_NOT_APPLICABLE!>aInString2 == aOutInt2<!>
+    <!EQUALITY_NOT_APPLICABLE!>aString == a2<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>bString == bInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>bOutString == bOutInt<!>
+    bInString == bInInt
+    <!EQUALITY_NOT_APPLICABLE!>bOutString == bInInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>bInString == bOutInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>bOutString == bInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>bInString == bInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>bOutString2 == bOutInt2<!>
+    bInString2 == bInInt2
+    <!EQUALITY_NOT_APPLICABLE!>bOutString2 == bInInt2<!>
+    <!EQUALITY_NOT_APPLICABLE!>bInString2 == bOutInt2<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>e == i<!>
+    <!EQUALITY_NOT_APPLICABLE!>"" == i<!>
+    <!EQUALITY_NOT_APPLICABLE!>ac == ad<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>tSub1 == tSub2<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>aString == bString<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>aListInt == aSetInt<!>
+    <!EQUALITY_NOT_APPLICABLE!>aSetInt == aListString<!>
+    <!EQUALITY_NOT_APPLICABLE!>aListString == aListInt<!>
+
+    <!EQUALITY_NOT_APPLICABLE!>aString == aListString<!>
+    <!EQUALITY_NOT_APPLICABLE!>bString == aListString<!>
```

Comment:
I'm not sure how to interpret these differences... actually, I don't understand what made these differences?!

Reply:
I am not sure I understand your question. The new implementation flags this comparison as invalid in the extended mode because `B` and `A` are unrelated classes.

Reply:
Duh, I totally missed `ConeTypeCompatibilityChecker.kt` which is collapsed by default!